### PR TITLE
Fix for #1719.

### PR DESCRIPTION
**Pull request by DeepakRajendrakumaran**

### CODE REVIEW

File: src/expr.cpp [hunk 1]
```diff
@@ -2034,12 +2034,21 @@ static ConstExpr *lConstFoldBinaryIntOp(BinaryExpr::Op op, const T *v0, const T
...
-        FOLD_OP_REF(BinaryExpr::Mod, %, TRef);
         FOLD_OP_REF(BinaryExpr::Shl, <<, TRef);
         FOLD_OP_REF(BinaryExpr::Shr, >>, TRef);
         FOLD_OP_REF(BinaryExpr::BitAnd, &, TRef);
         FOLD_OP_REF(BinaryExpr::BitXor, ^, TRef);
         FOLD_OP_REF(BinaryExpr::BitOr, |, TRef);
+    case BinaryExpr::Mod:
+        for (int i = 0; i < count; ++i) {
+            if (v1[i] == 0) {
+                Warning(pos, "Division by zero encountered in expression.");
```
Comment (dbabokin) [inline src/expr.cpp:2045]: It's not division, clang gives the following message: "warning: remainder by zero is undefined".
For division it has: "division by zero is undefined".

File: src/expr.cpp [hunk 2]
```diff
@@ -2086,10 +2095,11 @@ static ConstExpr *lConstFoldBinaryArithOp(BinaryExpr::Op op, const T *v0, const
     case BinaryExpr::Div:
```
Comment (dbabokin) [inline src/expr.cpp:2095]: Div is handled here, while mod in lConstFoldBinaryIntOp(). This is inconsistent. This needs to be handle in the same place (either of them).

File: src/expr.cpp (hunk 1 above)
Comment (DeepakRajendrakumaran) [reply src/expr.cpp:2045]: Whoops..will change it - I checked with gcc : https://godbolt.org/z/WYMXnn .
Clang was different - https://godbolt.org/z/D9M9tw

File: src/expr.cpp (hunk 2 above)
Comment (DeepakRajendrakumaran) [reply src/expr.cpp:2095]: That'd by design
lConstFoldBinaryIntOp : Constant fold the binary integer operations that aren't also applicable  to floating-point types. So, Mod goes here

lConstFoldBinaryArithOp - For those applicable for float. Hence div

File: src/expr.cpp (hunk 1 above)
Comment (DeepakRajendrakumaran) [reply src/expr.cpp]: Done

File: src/expr.cpp (hunk 2 above)
Comment (dbabokin) [reply src/expr.cpp:2095]: Ok, looks good.

**Pull request merged**